File: r/sample.r
# Define the transactions as a data frame
transactions <- data.frame(
  transactionId = c(1, 2, 3, 4),
  amount = c(1000, 500, 1500, 700),
  transactionDate = as.Date(c("2023-01-01", "2023-01-02", "2023-01-03", "2023-01-04")),
  transactionType = c("income", "expense", "income", "expense")
)

# Function to calculate net profit
calculateNetProfit <- function(transactions, taxes) {
  # Calculate total income
  totalIncome <- sum(transactions$amount[transactions$transactionType == "income"])
  
  # Calculate total expenses
  totalExpenses <- sum(transactions$amount[transactions$transactionType == "expense"])
  
  # Calculate net income
  netIncome <- totalIncome - totalExpenses
  
  # Calculate net profit
  netProfit <- netIncome - taxes
  
  return(netProfit)
}

# Example usage
taxes <- 200
netProfit <- calculateNetProfit(transactions, taxes)
print(paste("Net Profit: ", netProfit))

File: php/sample.php
<?php

class Transaction {
    public int $transactionId;
    public float $amount;
    public string $transactionDate;
    public string $transactionType;  // 'income' or 'expense'

    public function __construct(int $transactionId, float $amount, string $transactionDate, string $transactionType) {
        $this->transactionId = $transactionId;
        $this->amount = $amount;
        $this->transactionDate = $transactionDate;
        $this->transactionType = $transactionType;
    }
}

function calculateNetProfit(array $transactions, float $taxes): float {
    $totalIncome = 0.0;
    $totalExpenses = 0.0;

    foreach ($transactions as $transaction) {
        if ($transaction->transactionType === 'income') {
            $totalIncome += $transaction->amount;
        } elseif ($transaction->transactionType === 'expense') {
            $totalExpenses += $transaction->amount;
        }
    }

    $netIncome = $totalIncome - $totalExpenses;
    $netProfit = $netIncome - $taxes;
    return $netProfit;
}

// Example usage
$transactions = [
    new Transaction(1, 1000.0, '2023-01-01', 'income'),
    new Transaction(2, 500.0, '2023-01-02', 'expense'),
    new Transaction(3, 1500.0, '2023-01-03', 'income'),
    new Transaction(4, 700.0, '2023-01-04', 'expense')
];

$taxes = 200.0;
$netProfit = calculateNetProfit($transactions, $taxes);
echo "Net Profit: $netProfit\n";
?>
